File: leetcode/0383-ransom-note/solution.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        counter = [0] * 26

        if len(ransomNote) > len(magazine):
            return False

        for ch in magazine:
            counter[ord(ch) - ord('a')] += 1

        for ch in ransomNote:
            idx = ord(ch) - ord('a') 
            if counter[idx] == 0:
                return False
            counter[idx] -= 1
        
        return True

File: leetcode/0383-ransom-note/solution.rs
use std::collections::HashSet;

impl Solution {
    pub fn can_construct(ransom_note: String, magazine: String) -> bool {
        if ransom_note.len() > magazine.len() {
            return false;
        }

        let mut count = [0; 26];

        for b in magazine.bytes() {
            count[(b - b'a') as usize] += 1;
        }

        for b in ransom_note.bytes() {
            let idx = (b - b'a') as usize;
            match count[idx] {
                0 => return false,
                _ => count[idx] -= 1, 
            }
        }

        true   
    }
}
